tanner/config.py

```diff
@@ -1,92 +1,52 @@
-                   'EMULATOR_ENABLED': {'sqli': True, 'rfi': True, 'lfi': True, 'xss': True, 'cmd_exec': True,
-                                        'php_code_injection': True, 'php_object_injection': True, "crlf": True,
-                                        "xxe_injection": True, "template_injection": True},
-                   'SQLI': {'type': 'SQLITE', 'db_name': 'tanner_db', 'host': 'localhost', 'user': 'root',
-                            'password': 'user_pass'},
-                   'XXE_INJECTION': {'OUT_OF_BAND': False},
-                   'RFI': {"allow_insecure": False},
-                   'DOCKER': {'host_image': 'busybox:latest'},
-                   'LOGGER': {'log_debug': '/opt/tanner/tanner.log', 'log_err': '/opt/tanner/tanner.err'},
-                   'MONGO': {'enabled': False, 'URI': 'mongodb://localhost'},
-                   'HPFEEDS': {'enabled': False, 'HOST': 'localhost', 'PORT': 10000, 'IDENT': '', 'SECRET': '',
-                               'CHANNEL': 'tanner.events'},
-                   'LOCALLOG': {'enabled': False, 'PATH': '/tmp/tanner_report.json'},
-                   'CLEANLOG': {'enabled': False},
-                   'REMOTE_DOCKERFILE': {'GITHUB': "https://raw.githubusercontent.com/mushorg/tanner/master/docker/"
-                                                   "tanner/template_injection/Dockerfile"},
-                   'SESSIONS': {"delete_timeout": 300}
-                   }
+
+def read_config(path):
+    config_values = {}
+    with open(path, 'r') as f:
+        config_values = yaml.load(f, Loader=yaml.FullLoader)
+    return config_values
+
+
+DEFAULT_CONFIG = read_config("/opt/tanner/data/config.yaml")
 
 
 class TannerConfig():
     config = None
 
     @staticmethod
     def set_config(config_path):
-        cfg = configparser.ConfigParser()
         if not os.path.exists(config_path):
-            print("Config file {} doesn't exist. Check the config path or use default".format(config_path))
+            print("Config file {} doesn't exist. Check the config path or use default".format(
+                config_path))
             sys.exit(1)
 
-        cfg.read(config_path)
-        TannerConfig.config = cfg
+        TannerConfig.config = read_config(config_path)
 
     @staticmethod
     def get(section, value):
-        res = None
         if TannerConfig.config is not None:
             try:
-                convert_type = type(config_template[section][value])
-                if convert_type is bool:
-                    res = TannerConfig.config.getboolean(section, value)
-                else:
-                    res = convert_type(TannerConfig.config.get(section, value))
-            except (configparser.NoOptionError, configparser.NoSectionError):
-                LOGGER.warning("Error in config, default value will be used. Section: %s Value: %s", section, value)
-                res = config_template[section][value]
-
+                res = TannerConfig.config[section][value]
```

Comment:
instead of using if/try you can use noly try and catch also TypeError (`case TannerConfig.config is None`)

or

you can merge together default config and user config